x/liquidity/keeper/liquidity_pool.go

```diff
@@ -240,77 +237,53 @@ func (k Keeper) ExecuteDeposit(ctx sdk.Context, msg types.DepositMsgState, batch
 		return nil
 	}
 
-	// only two coins are acceptable
-	if reserveCoins.Len() != msg.Msg.DepositCoins.Len() {
-		return types.ErrNumOfReserveCoin
-	}
-
 	reserveCoins.Sort()
 
-	// Decimal Error, divide the Int coin amount by the Decimal Rate and erase the decimal point to deposit a lower value
-	lastReserveCoinA := reserveCoins[0].Amount
-	lastReserveCoinB := reserveCoins[1].Amount
-	lastReserveRatio := lastReserveCoinA.ToDec().QuoTruncate(lastReserveCoinB.ToDec())
+	lastReserveCoinA := reserveCoins[0]
+	lastReserveCoinB := reserveCoins[1]
 
 	depositCoinA := depositCoins[0]
 	depositCoinB := depositCoins[1]
-	depositCoinAmountA := depositCoinA.Amount
-	depositCoinAmountB := depositCoinB.Amount
-	depositableCoinAmountA := depositCoinB.Amount.ToDec().MulTruncate(lastReserveRatio).TruncateInt()
-
-	refundedCoins := sdk.NewCoins()
-	refundedCoinA := sdk.ZeroInt()
-	refundedCoinB := sdk.ZeroInt()
 
-	var acceptedCoins sdk.Coins
-	// handle when depositing coin A amount is less than, greater than or equal to depositable amount
-	if depositCoinA.Amount.LT(depositableCoinAmountA) {
-		depositCoinAmountB = depositCoinA.Amount.ToDec().QuoTruncate(lastReserveRatio).TruncateInt()
-		acceptedCoins = sdk.NewCoins(depositCoinA, sdk.NewCoin(depositCoinB.Denom, depositCoinAmountB))
-
-		inputs = append(inputs, banktypes.NewInput(batchEscrowAcc, acceptedCoins))
-		outputs = append(outputs, banktypes.NewOutput(reserveAcc, acceptedCoins))
-
-		refundedCoinB = depositCoinB.Amount.Sub(depositCoinAmountB)
-
-		if refundedCoinB.IsPositive() {
-			refundedCoins = sdk.NewCoins(sdk.NewCoin(depositCoinB.Denom, refundedCoinB))
-			inputs = append(inputs, banktypes.NewInput(batchEscrowAcc, refundedCoins))
-			outputs = append(outputs, banktypes.NewOutput(depositor, refundedCoins))
-		}
-	} else if depositCoinA.Amount.GT(depositableCoinAmountA) {
-		depositCoinAmountA = depositCoinB.Amount.ToDec().MulTruncate(lastReserveRatio).TruncateInt()
-		acceptedCoins = sdk.NewCoins(depositCoinB, sdk.NewCoin(depositCoinA.Denom, depositCoinAmountA))
-
-		inputs = append(inputs, banktypes.NewInput(batchEscrowAcc, acceptedCoins))
-		outputs = append(outputs, banktypes.NewOutput(reserveAcc, acceptedCoins))
+	poolCoinTotalSupply := k.GetPoolCoinTotalSupply(ctx, pool)
+	poolCoinMintAmt := sdk.MinDec(
+		poolCoinTotalSupply.ToDec().MulTruncate(depositCoinA.Amount.ToDec()).QuoTruncate(lastReserveCoinA.Amount.ToDec()),
+		poolCoinTotalSupply.ToDec().MulTruncate(depositCoinB.Amount.ToDec()).QuoTruncate(lastReserveCoinB.Amount.ToDec()),
+	)
+	mintRate := poolCoinMintAmt.TruncateDec().QuoTruncate(poolCoinTotalSupply.ToDec())
+	acceptedCoins := sdk.NewCoins(
+		sdk.NewCoin(depositCoins[0].Denom, lastReserveCoinA.Amount.ToDec().Mul(mintRate).TruncateInt()),
+		sdk.NewCoin(depositCoins[1].Denom, lastReserveCoinB.Amount.ToDec().Mul(mintRate).TruncateInt()),
```

Comment:
```suggestion
		sdk.NewCoin(depositCoins[0].Denom, lastReserveCoinA.Amount.ToDec().MulTruncate(mintRate).TruncateInt()),
		sdk.NewCoin(depositCoins[1].Denom, lastReserveCoinB.Amount.ToDec().MulTruncate(mintRate).TruncateInt()),
```
What do you think about using `MulTruncate` instead of `Mul` here? @hallazzang @typark391 @kogisin 
I think both will have the same result but that is more intuitive, or is there a case that is not?

Reply:
Using `MulTruncate` here can cause(very rarely) a depositor to take advantage against the pool.
Since the result of `MulTruncate` is always less or equal than the result of `Mul`, I suggest to use `Mul` here.
How do others think?

Reply:
That's right. Let's keep it `Mul` first, reproduce the corresponding edge case, analyze it, and specify it on spec through other PRs.